File: 7-reverse-integer/reverse-integer.java
class Solution {
    public int reverse(int x) {
        long rev = 0;           // Use long instead of int
        int original = x;
        
        while (x != 0) {
            rev = rev * 10 + x % 10;
            x /= 10;
        }
        
        // Cast back to int at the end
        if (rev > Integer.MAX_VALUE || rev < Integer.MIN_VALUE) {
            return 0;
        }
        
        return (int) rev;
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, n: int) -> int:
        x = abs(n)
        rev =0

        while(x>0):
            dig = x%10
            rev=rev*10+dig
            x//=10
        if(rev>=(2**31)-1): return 0
        if n<0 :
            
            return -1*rev
        return rev
        